src/components/MapView/Legends/exposedPopulationAnalysis.tsx

```diff
@@ -80,20 +83,49 @@ const ExposedPopulationAnalysis = ({
     await dispatch(requestAndStoreExposedPopulation(params));
   };
 
-  const ResultSwitches = () => (
-    <FormGroup>
-      <AnalysisFormControlLabel
-        control={
-          <Switch
-            color="primary"
-            checked={isMapLayerActive}
-            onChange={e => dispatch(setIsMapLayerActive(e.target.checked))}
+  const ResultSwitches = () => {
+    const data = useSelector(analysisResultSelector);
+    const features = data?.featureCollection.features;
+    const hasData = features?.length === 0 || false;
+
+    const handleTableViewChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      dispatch(setIsDataTableDrawerActive(e.target.checked));
+      dispatch(
+        setCurrentDataDefinition({
+          id: TableDefinitions.animal_deaths.id,
+          title: data?.getTitle() as string,
+          table: '',
+          legendText: data?.legendText as string,
```

Comment:
it isn't great to cast like this and ignore TS's concerns that these could be undefined. Maybe we can do an if check to make sure they are indeed not undefined? Otherwise, if we're 100% certain these can't be undefined we can leave as is

Reply:
You are correct. There is potential for it to be `undefinded`. Solved with implementation below

```javascript
{
          id: TableDefinitions.animal_deaths.id,
          title: data ? data.getTitle() : '',
          table: '',
          legendText: data ? data.legendText : '',
}
```